pkg/apiserver/resthandler.go

```diff
@@ -299,6 +301,27 @@ func createHandler(r rest.NamedCreater, scope RequestScope, typer runtime.Object
 			return
 		}
 
+		unstruct := &runtime.Unstructured{}
+		err = runtime.UnstructuredJSONScheme.DecodeInto(body, unstruct)
```

Comment:
It would be nice if instead we could pass apiVersion and kind from the URL to Decode and then verify later that apiVersion and kind from the object match.